src/Modules/OrchardCore.Commerce.ContentFields/Settings/PriceFieldSettingsDriver.cs

```diff
@@ -0,0 +1,82 @@
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.Extensions.Localization;
+using OrchardCore.Commerce.ContentFields.Models;
+using OrchardCore.Commerce.ContentFields.Settings;
+using OrchardCore.Commerce.ContentFields.ViewModels;
+using OrchardCore.Commerce.MoneyDataType.Abstractions;
+using OrchardCore.ContentManagement.Metadata.Models;
+using OrchardCore.ContentTypes.Editors;
+using OrchardCore.DisplayManagement.ModelBinding;
+using OrchardCore.DisplayManagement.Views;
+using System;
+using System.Threading.Tasks;
+
+namespace OrchardCore.Commerce.Settings;
+
+public class PriceFieldSettingsDriver : ContentPartFieldDefinitionDisplayDriver<PriceField>
+{
+    private readonly IStringLocalizer<PriceFieldSettingsDriver> T;
+    private readonly IMoneyService _moneyService;
+
+    public PriceFieldSettingsDriver(IStringLocalizer<PriceFieldSettingsDriver> localizer, IMoneyService moneyService)
+    {
+        T = localizer;
+        _moneyService = moneyService;
+    }
+
+    public override IDisplayResult Edit(ContentPartFieldDefinition model, IUpdateModel updater) =>
+        Initialize($"{nameof(PriceFieldSettings)}_Edit", (Action<PriceFieldSettingsEditViewModel>)(viewModel =>
+            {
+                var settings = model.GetSettings<PriceFieldSettings>();
+
+                viewModel.Hint = settings.Hint;
+                viewModel.Label = settings.Label;
+                viewModel.Required = settings.Required;
+
+                viewModel.CurrencySelectionMode = settings.CurrencySelectionMode;
+                viewModel.CurrencySelectionModes = new SelectList(
+                    new[]
+                    {
+                        new CurrencySelectionModeItem(CurrencySelectionMode.AllCurrencies, T["All Currencies"]),
+                        new CurrencySelectionModeItem(CurrencySelectionMode.DefaultCurrency, T["Default Currency"]),
+                        new CurrencySelectionModeItem(CurrencySelectionMode.SpecificCurrency, T["Specific Currency"]),
+                    },
+                    nameof(CurrencySelectionModeItem.Value),
+                    nameof(CurrencySelectionModeItem.Text),
+                    (int)settings.CurrencySelectionMode);
+
+                viewModel.SpecificCurrencyIsoCode = settings.SpecificCurrencyIsoCode;
+                viewModel.Currencies = _moneyService.GetSelectableCurrencies(
+                    CurrencySelectionMode.AllCurrencies,
+                    localizer: T,
+                    currentValue: settings.SpecificCurrencyIsoCode);
+            }))
+            .Location("Content");
+
+    public override async Task<IDisplayResult> UpdateAsync(ContentPartFieldDefinition model, UpdatePartFieldEditorContext context)
```

Comment:
Why is that preferable?

Reply:
For consistency, I mean I just don't understand why is this different than the one in PartFieldDisplayDriver.cs. But you wrote in a previous comment that you have copied that one from OrchardCore.ContentFields, so I guess that's why.

Reply:
They are different types, and `ContentPartFieldDefinitionDisplayDriver<TField>` doesn't have an override that contains both the context and the updater, just either of the two:
![image](https://user-images.githubusercontent.com/4007293/188509916-9d8caada-6ba8-445c-a6c7-d9a6951747a0.png)
So using context is the more inclusive option since I need `context.Builder` too.

Reply:
Ah, okay!